z.lua

```diff
@@ -1460,6 +1462,72 @@ function z_shell_init(opts)
 	end
 end
 
+-----------------------------------------------------------------------
+-- Fish shell init
+-----------------------------------------------------------------------
+local script_zlua_fish = [[
+function _zlua
+	argparse --name 'z' -x 't,r' -x 'l,e,x'\
+	l e x t r c s i h/help A-add C-complete -- $argv
+
+	if test -n "$_flag_add"
+		env _ZL_RANDOM="$RANDOM" "$ZLUA_LUAEXE" "$ZLUA_SCRIPT" --add "$argv"
+		return
+	else if test -n "$_flag_complete"
+		"$ZLUA_LUAEXE" "$ZLUA_SCRIPT" --complete "$argv"
+		return
+	end
+
+	set -l arg_type "$_flag_t$_flag_r"
+	set -l arg_mode "$_flag_l$_flag_e$_flag_x"
+
+	if test  -n "$_flag_h"
+		"$ZLUA_LUAEXE" "$ZLUA_SCRIPT" -h
+	else if echo $arg_type | string match '*f'; or test (count $argv) -eq 0; # <- not needed
+		"$ZLUA_LUAEXE" "$ZLUA_SCRIPT" -l $_flag_c $arg_type $_flag_s "$argv"
+	else if test -n "$arg_mode"
```

Comment:
can we use square brackets in fish shell instead of test ?? like:

    if [ -n "$arg_mode ]

??

Reply:
test is more idiomatic for Fish

Reply:
fine